src/components/encounter/ohri-encounter-form.component.tsx

```diff
@@ -484,7 +485,22 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
 
     if (encounterForSubmission.obs?.length || encounterForSubmission.orders?.length) {
       const ac = new AbortController();
-      return saveEncounter(ac, encounterForSubmission, encounter?.uuid);
+      return saveEncounter(ac, encounterForSubmission, encounter?.uuid)
+        .then(response => response.json())
+        .then(encounter => {
+          fields
+            ?.filter(field => field?.questionOptions.rendering === 'file')
+            .forEach(field => {
+              saveAttachment(
+                encounter.patient.uuid,
+                field?.value?.value,
+                field?.questionOptions.concept,
+                new Date().toISOString(),
+                encounter.uuid,
+                ac,
+              );
+            });
+        });
```

Comment:
I think we can proceed this way for now though this makes the encounter component aware of "attachments". I can think of two ways of approaching this:

1. Comprehensive support for `FormProcessors`. This kind of flow would be configured at the processor level
2. A special inbuilt post-submission handler. It's probably not as ideal but it's configured through the registry. It wouldn't introduce this level of coupling.

---

@arodidev can you resolve the attachment specific Promises in parallel?

Reply:
@samuelmale a few questions.
1. Isn't the encounter supposed to be aware of the attachment, for the purposes of tying the attachment to a specific encounter?
2. Could you kindly expound on the comment regarding the implementation approaches, and also regarding the attachment Promises?

Reply:
> Isn't the encounter supposed to be aware of the attachment, for the purposes of tying the attachment to a specific encounter? 

This is correct, we wanna link the complex obs to the current encounter, my concerns were around the means of achieving this. I'm sold to an abstract way of solving the problem. 

> Could you kindly expound on the comment regarding the implementation approaches, and also regarding the attachment Promises?

```js

.then(encounter => {
    const attachmentPromises = fields
      ?.filter(field => field?.questionOptions.rendering === 'file')
      .map(field => {
        return saveAttachment(
          encounter.patient.uuid,
          field?.value?.value,
          field?.questionOptions.concept,
          new Date().toISOString(),
          encounter.uuid,
          ac,
        );
      });

    return Promise.all(attachmentPromises)
      .then(() => {
        // Any further code to execute after all attachments are saved.
      });
  });
  
```

Reply:
Understood